aws/signer/v4/v4.go

```diff
@@ -344,7 +345,14 @@ func (s *Signer) PresignHTTP(
 
 	logSigningInfo(ctx, options, &signedRequest, true)
 
-	return signedRequest.Request.URL.String(), signedRequest.SignedHeaders, nil
+	signedHeaders = make(http.Header)
+
+	for k, v := range signedRequest.SignedHeaders {
+		key := textproto.CanonicalMIMEHeaderKey(k)
+		signedHeaders[key] = append(signedHeaders[key], v...)
+	}
```

Comment:
would be helpful to have a comment here calling out why this is done, vs just the request's original-cased header keys.